File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/ui/markers/base/MarkerPresenter.java
package com.example.ama.android2_lesson03.ui.markers.base;

import com.example.ama.android2_lesson03.repo.data.model.SimpleMarker;
import com.example.ama.android2_lesson03.ui.base.Presenter;

public interface MarkerPresenter<T extends MarkerView> extends Presenter<T> {
    void getMarkerList();

    void tuneMapCurrentMarker(SimpleMarker marker);

    void editMarkerName(SimpleMarker marker, String newName);

    void deleteMarker(SimpleMarker marker);

    void onUpdateMarker(SimpleMarker marker);

}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/ui/markers/base/MarkerPresenter.kt
package com.example.ama.android2_lesson03.ui.markers.base

import com.example.ama.android2_lesson03.repo.data.model.SimpleMarker
import com.example.ama.android2_lesson03.ui.base.Presenter

interface MarkerPresenter<T : MarkerView> : Presenter<T> {
    fun getMarkerList()
    fun tuneMapCurrentMarker(marker: SimpleMarker)
    fun editMarkerName(marker: SimpleMarker, newName: String)
    fun deleteMarker(marker: SimpleMarker)
    fun onUpdateMarker(marker: SimpleMarker)
}